### ISSUE TITLE
Does the pair embedding break SE(3) equivariance because it uses global direction vectors?

### ISSUE THREAD

**Issue by GLX1999** [alexjli/proteinzen#2]
Hi,
In proteinzen/model/modules/pair_modules.py, _featurize_rigids concatenates [quat_rel, unit_edge_dist_vecs, dist_features]. quat_rel = q_i^{-1} q_j and the distance encodings are SE(3)-invariant, but unit_edge_dist_vecs = F.normalize(X_i - X_j) is expressed in the global coordinate frame and rotates under a rigid-body rotation. Since this direction feature is then passed through a standard linear layer (self.lin_a_ij) into the pair stack / IPA pair bias, does a global rotation change attention logits and thus make the overall model not strictly SE(3)-equivariant? Is this an intended design choice, or should the direction feature be represented in a relative/local frame to preserve equivariance?

**Comment by alexjli** [comment]
You're totally right, the direction vectors are suposed to be represented in the local frame, this seems to be a bug... Thanks for catching it! The current set of pretrained weights were trained and benchmarked with the bug so I won't hot-fix it for this current release but I'll be sure to fix it in the next update.
